cmd/influx/telegraf.go

```diff
@@ -0,0 +1,334 @@
+	}
+
+	cfgs, _, err := svc.FindTelegrafConfigs(context.Background(), influxdb.TelegrafConfigFilter{
+		OrgID:                     &orgID,
+		UserResourceMappingFilter: filter,
+	})
+	if err != nil {
+		return err
+	}
+	return b.writeTelegrafConfig(cfgs...)
+}
+
+func (b *cmdTelegrafBuilder) cmdCreate() *cobra.Command {
+	cmd := b.newCmd("create", b.createRunEFn)
+	cmd.Short = "Create a telegraf configuration"
+	cmd.Long = `
+	The telegrafs create command creates a new telegraf configuration.
+
+	Examples:
+		# create new telegraf configuration
+		influx telegrafs create --name $CFG_NAME --description $CFG_DESC --file $PATH_TO_TELE_CFG
+
+		# create new telegraf configuration using shorts
+		influx telegrafs create -n $CFG_NAME -d $CFG_DESC -f $PATH_TO_TELE_CFG
+`
+
+	b.org.register(cmd, false)
+	b.registerTelegrafCfgFlags(cmd)
+
+	return cmd
+}
+
+func (b *cmdTelegrafBuilder) createRunEFn(cmd *cobra.Command, args []string) error {
+	svc, orgSVC, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	orgID, err := b.org.getID(orgSVC)
+	if err != nil {
+		return err
+	}
+
+	cfg, err := b.readConfig(b.file)
+	if err != nil {
+		return err
+	}
+
+	newTelegraf := influxdb.TelegrafConfig{
+		OrgID:       orgID,
+		Name:        b.name,
+		Description: b.desc,
+		Config:      cfg,
+	}
+	err = svc.CreateTelegrafConfig(context.Background(), &newTelegraf, 0)
+	if err != nil {
+		return err
+	}
+
+	return b.writeTelegrafConfig(&newTelegraf)
+}
+
+func (b *cmdTelegrafBuilder) cmdRemove() *cobra.Command {
+	cmd := b.newCmd("rm", b.removeRunEFn)
+	cmd.Aliases = []string{"remove"}
+	cmd.Short = "Remove telegraf configuration(s)"
+	cmd.Long = `
+	The telegrafs rm command removes telegraf configuration(s).
+
+	Examples:
+		# remove a single telegraf configuration
+		influx telegrafs rm --id $ID
+
+		# remove multiple telegraf configurations
+		influx telegrafs rm --id $ID1 --id $ID2
+
+		# remove using short flags
+		influx telegrafs rm -i $ID1
+`
+
+	cmd.Flags().StringArrayVarP(&b.ids, "id", "i", nil, "Telegraf configuration id(s) to remove")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func (b *cmdTelegrafBuilder) removeRunEFn(cmd *cobra.Command, args []string) error {
+	svc, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	for _, rawID := range b.ids {
+		id, err := influxdb.IDFromString(rawID)
+		if err != nil {
+			return err
+		}
+
+		err = svc.DeleteTelegrafConfig(context.Background(), *id)
+		if err != nil && influxdb.ErrorCode(err) != influxdb.ENotFound {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (b *cmdTelegrafBuilder) cmdUpdate() *cobra.Command {
+	cmd := b.newCmd("update", b.updateRunEFn)
+	cmd.Short = "Update a telegraf configuration"
+	cmd.Long = `
+	The telegrafs update command udpates a new telegraf configuration. The update will update
+	the telegraf configuration to exactly match the provided parameters to this command. If a
+	name or description is not given, then it will be set to an empty string.
```

Comment:
The telegrafs update command updates a Telegraf configuration to match the specified parameters.
If a name or description is not provided, then are set to an empty string.